src/components/VulnerabilityEdgePrompt.js

```diff
@@ -1,39 +1,174 @@
 import React from 'react';
-import {
-  Button,
-  EmptyState,
-  EmptyStateBody,
-  EmptyStateIcon,
-  Title,
-} from '@patternfly/react-core';
+import { useContext, useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { MonitoringIcon } from '@patternfly/react-icons';
-import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import RegistryContext from '../store/registeryContext';
+import { Alert, Button } from '@patternfly/react-core';
+import AsyncComponent from '@redhat-cloud-services/frontend-components/AsyncComponent';
+import { getDevice, getImageSet } from '../api/edge/imagesInfo';
+import PropTypes from 'prop-types';
 
+import { InProgressIcon } from '@patternfly/react-icons';
 const VulnerabilityEdgePrompt = () => {
-  const chrome = useChrome();
+  const { getRegistry } = useContext(RegistryContext);
   const { inventoryId } = useParams();
+  const [CVEs, setCVEs] = useState(null);
+  const [imageId, setImageID] = useState(null);
+  const [imageSetId, setImageSetID] = useState(null);
+  const [deviceData, setDeviceData] = useState(null);
+  const [newImageStatus, setNewImageStatus] = useState(null);
+  useEffect(() => {
+    if (CVEs?.data?.length > 0) {
+      const data = getActiveAlert(CVEs, deviceData, newImageStatus, imageId);
+      setActiveAlert(data);
+    }
+  });
+  useEffect(() => {
+    (async () => {
+      const a = await getDevice(inventoryId);
```

Comment:
Can we rename this 'a' to something more meaningful? maybe something like deviceInfo? doesnt necissarly have to be that its just that 'a' doesnt mean anything